### ISSUE TITLE
any type with string causes error

### ISSUE THREAD

**Issue by qian81** [byu-oit/openapi-enforcer#31]
hello,

I'm using openapi-enforcer-middleware in my project. 
In my API definition, it contains property with any type, like:

```yml
properties:
  value:
    oneOf:
      - type: string
      - type: number
      - type: boolean
      - type: array
        items:
          oneOf:
            - type: string
            - type: number
            - type: boolean
            - type: object
      - type: object
```

If I sends a request e.g. value: "test" as string to my service, my controller function is not being called. And also no exception or error generated. I did some investigation and find out that in src/schema/serialize.js, Object.assign(value, highs[0].result) is not working with string value. 
 
I added try/catch around object.assign(...) and it generates the following:
TypeError: Cannot assign to read only property '0' of object '[object String]'

Could you please try to fix that, maybe with code as below?

if (typeof value == „string“) { value = matches[0].result } 
else { Object.assign(value, matches[0].result }

thanks a lot!

**Comment by Gi60s** [comment]
I see the problem. I can definitely get this fixed, although it might be Monday before I can get to it. Sorry about that.

**Comment by Gi60s** [comment]
This has been fixed with PR #33 and has been published to NPM under version `1.5.3`. Thanks for the bug submit.
